solution.md

```diff
@@ -0,0 +1,179 @@
+# Problem 1 - Range Sum of BST
+## Time Complexity :
+O(n)
+
+## Space Complexity :
+O(n)
+
+## Did this code successfully run on Leetcode :
+Yes.
+
+## Any problem you faced while coding this :
+No.
+
+## Your code here along with comments explaining your approach
+### Solution 1: DFS
+      class Solution:
+          def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
+              def dfs(root: TreeNode, L: int, R: int):
+                  if not root:
+                      return 0
+                  #Case1
+                  if root.val>=L and root.val<=R:
+                      return dfs(root.left,L,R) + root.val + dfs(root.right,L,R)
+                  #Case2
+                  elif root.val<L:
+                      return dfs(root.right,L,R)
+                  #Case3
+                  return dfs(root.left,L,R)
+
+              return dfs(root, L, R)
+
+# Problem 1 - Range Sum of BST
+## Time Complexity :
+O(n)
+
+## Space Complexity :
+O(n)
+
+## Did this code successfully run on Leetcode :
+No - Time Limit Exceeded.
+
+## Any problem you faced while coding this :
+Yes. Cannot figure out why the code is not running.
+
+## Your code here along with comments explaining your approach
+### Solution 2: BFS
+      class Solution:
+          def rangeSumBST(self, root: TreeNode, L: int, R: int) -> int:
+              if not root:
+                  return 0
+              queue = []
+              result = 0
+              queue.append(root)
+              while queue:
+                  root=queue[0]
```

Comment:
You are not popping the first element. *Rolling eye*. Change it to `queue.pop(0)` and it will work. Also note that pop(0) is of O(n) time hence it will push your code to O(n2) time.